Code Samples/Fib/main.cpp

```diff
@@ -1,58 +1,86 @@
+#include <thread>
+#include <vector>
+#include <cstdlib>
+#include <ctime>
+#include <cstring>
+#include <atomic>
+#include <chrono>
+#include <csignal>
+#include <unistd.h>  
 
 #include "thread.h"
 
 #define THREAD_COUNT 10
 
-static char block[] = "--block";
-int test = 0;
+static constexpr char block[] = "--block";
+static constexpr char crash[] = "--crash";
+static constexpr char test_flag[] = "--test";
+std::atomic<int> test_count{0};  // Thread-safe counter
+volatile std::sig_atomic_t g_signal_status = 0;
+
+void signal_handler(int signal) {
+    g_signal_status = signal;
+}
 
 int main(int argc, char **argv)
 {
-    srand(time(NULL));
+    std::signal(SIGINT, signal_handler);
 
-    static char pidText[] = "PID: ";
-    std::string helpText = "Attach a debugger and execute 'set foo=0' to continue";
-    char helloText[] = "Hello World!";
+    std::srand(static_cast<unsigned>(std::time(nullptr)));
 
-    std::cout << helloText << std::endl;
+    std::cout << "Hello World!" << std::endl;
 
-    pthread_t threads[THREAD_COUNT];
+    if (argc == 2) {
+        if (std::strcmp(block, argv[1]) == 0) {
+            std::cout << "Attach a debugger and set foo=0 to continue" << std::endl;
+            std::cout << "Process ID: " << getpid() << std::endl;
 
-    if (argc == 2 && !strcmp(block, argv[1]))
-    {
-        std::cout << helpText << std::endl;
-        volatile int foo = 1;
-        while (foo)
-            ;
+            volatile int foo = 1;
+            while (foo && g_signal_status == 0) {
+                std::this_thread::sleep_for(std::chrono::seconds(1));
+                std::cout << "Waiting... (press Ctrl-C to quit)" << std::endl;
+            }
+            return 0;
+        }
+        else if (std::strcmp(crash, argv[1]) == 0) {
+            std::cout << "Triggering intentional crash..." << std::endl;
+            volatile int foo = 0;
+            volatile int bar = 1 / foo;  
+            (void)bar;                   
+            return 1;                     
+        }
+        else if (std::strcmp(test_flag, argv[1]) == 0) {
+            std::cout << "Running in test mode" << std::endl;
+        }
     }
 
-    if (argc == 2 && !strcmp("--crash", argv[1]))
-    {
-        int foo = 0;
-        int bar = 1 / foo;
-    }
+    std::vector<std::thread> threads;
+    threads.reserve(THREAD_COUNT);
 
-    for (int i = 0; i < THREAD_COUNT; i++)
-    {
-        std::cout << "Test " << i << std::endl;
-        pthread_create(&threads[i], NULL, &thread_proc, NULL);
-    }
+    try {
+        for (int i = 0; i < THREAD_COUNT; ++i) {
+            std::cout << "Launching thread " << i << std::endl;
+            threads.emplace_back(thread_proc);
+        }
 
-    for (int i = 0; i < THREAD_COUNT; i++)
-    {
-        pthread_join(threads[i], NULL);
-        test++;
+        for (auto& t : threads) {
+            if (t.joinable()) {
+                t.join();
+                test_count.fetch_add(1, std::memory_order_relaxed);
```

Comment:
Is this `test_count` needed?

What do you expect might throw an exception here?  Thread creation?  If so, I'd suggest either use a try/catch around just the creation of the thread (std::thread is std::move-able, so can be moved into the vector).  Or, better yet, consider thread creation failure to be cause to terminate the process, and don't use a try/catch at all, for simplicity.